src/ray/common/scheduling/scheduling_test.cc

```diff
@@ -5,8 +5,106 @@
 #include <thread>
```

Comment:
These tests are great, but they're basically all integration tests for `getSchedulableNode`. Would also be great if we had some basic unit tests for the logic to update resources, check for violations, etc. A bit of refactoring as I suggested above should make this straightforward.

Reply:
Actually, there are lots of tests to check for violations (see `TEST_F(SchedulingTest, SchedulingTaskRequestTest)` in `scheduling_test.cc`. Also for updating node resource availability, see `TEST_F(SchedulingTest, SchedulingUpdateAvailableResourcesTest)`. Please let me know if you have other suggestions.